Tasks/ServiceFabricDeploy/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1

```diff
@@ -200,9 +203,21 @@ function Publish-UpgradedServiceFabricApplication
 
         $reg = Get-ServiceFabricApplicationType -ApplicationTypeName $names.ApplicationTypeName | Where-Object  { $_.ApplicationTypeVersion -eq $names.ApplicationTypeVersion }
         if ($reg)
-        {
-            Write-Host (Get-VstsLocString -Key SFSDK_UnregisteringExistingAppType -ArgumentList @($names.ApplicationTypeName, $names.ApplicationTypeVersion))
-            $reg | Unregister-ServiceFabricApplicationType -Force
+        {            
+            try {
+                Write-Host (Get-VstsLocString -Key SFSDK_UnregisteringExistingAppType -ArgumentList @($names.ApplicationTypeName, $names.ApplicationTypeVersion))            
+                $reg | Unregister-ServiceFabricApplicationType -Force    
+            }
+            catch [System.Fabric.FabricException] {
+                ## If SkipUpgrade param is set when existing version already deployed, then back out of upgrade
+                if($SkipUpgradeSameTypeAndVersion -And $Error[0].Exception.Message -like '*Application type and version is still in use*') {                    
+                    Write-Warning (Get-VstsLocString -Key SFSDK_SkipUpgradeWarning -ArgumentList @($reg.ApplicationTypeName, $reg.ApplicationTypeVersion))
+                    return
+                }
+                else {                    
+                    throw   
+                }                
+            }            
```

Comment:
Really don't know why the trailing spaces appear. I am just using VS Code on Win10 with default settings. The diff tool does not show them untill I actually remove them after a commit.